backend/corpora/common/utils/db_session.py

```diff
@@ -12,10 +12,11 @@
 
 
 class DBSessionMaker:
-    engine = create_engine(CorporaDbConfig().database_uri, connect_args={"connect_timeout": 5})
+
     _session_make = None
 
     def __init__(self):
+        self.engine = create_engine(CorporaDbConfig().database_uri, connect_args={"connect_timeout": 5})
```

Comment:
Only one engine needs to be built for the whole process. It was intentional but we can change to initialize once when the first `DBSessionMaker` is created.